mux/broadcaster.go

```diff
@@ -0,0 +1,98 @@
+/*
+Provides pubsub of messages over channels.
+A provider has a broadcaster into which it Submits messages and into
+which subscribers Register to pick up those messages.
+
+If one of the subscribers is not able to consume the message, then messages will be dropped for this consumer.
+It is possible to pass a function to handle dropped messages.
+
+*/
+package mux
+
+import (
+	"fmt"
+)
+
+type Broadcaster struct {
+	input   chan interface{}
+	reg     chan registration
+	unreg   chan unregistration
+	outputs map[chan<- interface{}]string
+}
+
+// Register a new channel to receive broadcasts
+func (b *Broadcaster) Register(consumerName string, newch chan<- interface{}) {
```

Comment:
Good job :)
Question: consumerName is only used to log error messages on backpressure, am I correct?

Reply:
yes, it is used only to identify the consumer that is dropping messages on backpressure

Reply:
:+1:

Reply:
I modified a bit the API to pass the backpressure behavior on subscription, so each subscriber can have a custom behavior, and there is not need to pass the name anymore